adduser.sh

```diff
@@ -30,18 +33,20 @@ function del_user() {
     local user=$1
     echo -n "Are you sure you want to delete user '${user}' with all files in ${HOME_DIR}/${user}? Enter 'y' for Yes and 'n' for No "
     read confirmation
-
-    local user_conf_record=$(grep ${user} ${USER_CONF})
+    message="${user} does not exists"
 
     if [ ${confirmation} == "y" ]; then
-        echo "Deleting user ${user}..."
-        if [ "${user_conf_record}" == "" ]; then
-            echo "user not found"
-        else
-            sed -i "/${user}/d" "${USER_CONF}"
-            rm -rf ${HOME_DIR}/${user}
-            echo "Successfully deleted user ${user}"
-        fi
+        COUNTER=1
+        while read user_conf; do
```

Comment:
Could this whole section be a single sed line?
```bash
sed -i "/^${user}:/d" "${USER_CONF}"
```
This section and the user add section seems like a very complex way of doing a regex on the first field.